src/crewai/crew.py

```diff
@@ -275,6 +291,29 @@ def set_private_attrs(self) -> "Crew":
 
         return self
 
+    @model_validator(mode="after")
+    def set_parent_flow(self, max_depth: int = 5) -> Optional["Crew"]:
+        """Find the nearest Flow instance in the call stack.
+
+        Args:
+            max_depth: Maximum frames to traverse up the call stack.
+
+        Returns:
+            The first Flow instance found in the call stack, or None.
+        """
+        stack = inspect.stack(context=0)[1 : max_depth + 1]
+        try:
+            for frame_info in stack:
+                candidate = frame_info.frame.f_locals.get("self")
+                if isinstance(candidate, Flow):
+                    self.parent_flow = candidate
+                    break
+            else:
+                self.parent_flow = None
+        finally:
+            del stack
```

Comment:
A few suggestions to make it a little more readable
```suggestion
        try:
            self.parent_flow = None
            for frame_info in stack:
                candidate = frame_info.frame.f_locals.get("self")
                if isinstance(candidate, Flow):
                    self.parent_flow = candidate
                    break
             return self   
        finally:
            del stack
```